modules/core/shared/src/main/scala/weaver/internals/TagExpr.scala

```diff
@@ -0,0 +1,100 @@
+package weaver.internals
+
+import cats.parse.{ Parser0 => P0, Parser => P }
+import cats.syntax.all.*
+
+sealed trait TagExpr {
+  def eval(tags: Set[String]): Boolean
```

Comment:
I'm not sure if `eval` is immediately intuitive. How about `matches` or `conforms`?

Reply:
Renamed to `matches`